sound/soc/sof/ipc4-pcm.c

```diff
@@ -457,18 +457,70 @@ static void ipc4_ssp_dai_config_pcm_params_match(struct snd_sof_dev *sdev, const
 	}
 }
 
+/*
+ * Fixup DAI link parameters for sampling rate based on
+ * DAI copier configuration.
+ */
+static int sof_ipc4_pcm_dai_link_fixup_rate(struct snd_sof_dev *sdev,
+					    struct snd_pcm_hw_params *params,
+					    struct sof_ipc4_copier *ipc4_copier)
+{
+	struct snd_interval *rate = hw_param_interval(params, SNDRV_PCM_HW_PARAM_RATE);
+	struct sof_ipc4_available_audio_format *available_fmt =
+		&ipc4_copier->available_fmt;
+	struct sof_ipc4_audio_format *audio_fmt =
+		&available_fmt->input_pin_fmts[0].audio_fmt;
+	unsigned int be_rate  = audio_fmt->sampling_frequency;
+	unsigned int fe_rate = params_rate(params);
+	bool fe_be_rate_match = false;
+	bool single_be_rate = true;
+	int i;
+
+	/*
```

Comment:
Not  sure if it helps or make any sense here, but:
```c
/* No need to check further if we have a match to begin with */
if (fe_rate == be_rate)
	return 0;
```

Reply:
@ujfalusi Wouldn't this be a bit misleading as we may have multiple be_rates? Sure this is correct, but the loop will get you the same result. I can add if this seems clearer of course.

Reply:
sure, the first iteration will test in the loop `if (be_rate == fe_rate)` and if this is true, then we break out and `if (!fe_be_rate_match)` would not be taken so we return with 0.

Reply:
@ujfalusi Didn't add the additional check now in V13.

Reply:
OK